library/src/layout/enum.rs

```diff
@@ -132,6 +132,27 @@ pub struct EnumElem {
     /// If set to `{auto}`, uses the spacing [below blocks]($func/block.below).
     pub spacing: Smart<Spacing>,
 
+    /// The horizontal alignment that enum numbers should have, or `{auto}` to
+    /// inherit the text alignment from the context.
+    ///
+    /// By default, this is set to `{end}`, which aligns enum numbers
+    /// towards end of the current text direction (in left-to-right script,
+    /// for example, this is the same as `{right}`). The choice of `{end}`
+    /// for horizontal alignment of enum numbers is usually preferred over
+    /// `{start}`, as numbers then grow away from the text instead of towards
+    /// it, avoiding certain visual issues. This option lets you override this
+    /// behavior, however.
+    ///
+    /// ````example
+    /// #set enum(number-align: start)
+    ///
+    /// 1.  Numbers here are start-aligned,
+    /// 9.  overriding the default of
+    /// 10. end-aligned.
+    /// ````
+    #[default(Smart::Custom(GenAlign::End))]
+    pub number_align: Smart<GenAlign>,
```

Comment:
would be better if this fails early for vertical alignments, see `TextDir` for an example on how to do that

Reply:
Thanks for the suggestion. I implemented `HorizontalAlign` and `VerticalAlign` structs for this in the source `align.typ` to avoid duplicating code (I'd also use this with `raw.rs` in my other PR), is that fine? Or should I restrict `HorizontalAlign` to `enum.rs` (and not add `VerticalAlign`)?